### PR TITLE
Replace BigInts with Logarithms

### PR DESCRIPTION
**Pull request by etnpce**
This gives a decent speedup post e305 xp, and increases the iterations in `increaseXp`.

This does cause a small slowdown (~10%) in level gains past e305 xp, due to the removal of some aggresive rounding in the players favor while converting to BigInt.

This also fixes an issue that the BigInt xp is not reset on rebirth, which I couldn't notify you of elsewhere because the project doesn't have an Issues tab, and I'm not making a pull request for a single line of code

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -1259,6 +1258,14 @@ function makeHeroes() {
 function assignMethods() {
     for (const key in gameData.taskData) {
         let task = gameData.taskData[key]
+
+        if (!('xpLog' in task) && ('xpBigInt' in task)){
+            if (typeof task.xpBigInt === "string" && task.xpBigInt.includes("e"))
+                task.xpBigInt = exponentialToRawNumberString(task.xpBigInt)
+
+            task.xpLog = task.xpBigInt.length + Math.log10("0." + task.xpBigInt.substring(0, 15)) // This line from StackOverflow
```
Comment (James103) [inline js/main.js:1266]: Which StackOverflow post did you take this line from?

Comment (etnpce) [reply js/main.js:1266]: I pulled that line from the top answer of https://stackoverflow.com/questions/70382306/logarithm-of-a-bigint

**Comment by thomaspj10** [comment]
Will probably merge this one, but I'm going to do some testing first.

**Comment by indomit** [comment]
Somewhere there is a bug in which XP is increased during an active pause for a while, and then when unpaused, progress freezes until the page is refreshed

**Comment by thomaspj10** [comment]
This is quite a big change which should probably get tested thoroughly

**Pull request merged**